core/iwasm/aot/aot_loader.c

```diff
@@ -302,7 +302,7 @@ loader_mmap(uint32 size, bool prot_exec, char *error_buf, uint32 error_buf_size)
     int map_flags;
     void *mem;
 
-#if UINTPTR_MAX == UINT64_MAX
+#if UINTPTR_MAX == UINT64_MAX && (!defined(__APPLE__)  || !defined(__arm64__))
```

Comment:
The `MMAP_MAP_32BIT` is only handled when the target is x86-64 or riscv64, and `__APPLE__` isn't defined. See posix os_mmap:
https://github.com/bytecodealliance/wasm-micro-runtime/blob/main/core/shared/platform/common/posix/posix_memmap.c#L83-L98

How about changing to:
```C
#if defined(BUILD_TARGET_X86_64) || defined(BUILD_TARGET_AMD_64) \
    || defined(BUILD_TARGET_RISCV64_LP64D)                       \
    || defined(BUILD_TARGET_RISCV64_LP64)
#ifndef __APPLE__
...
#endif
#endif
```

Reply:
Makes sense to me. Updated now.